cpp/arcticdb/version/version_store_api.cpp

```diff
@@ -1009,40 +1009,17 @@ std::pair<VersionedItem, py::object> PythonVersionStore::read_metadata(
     return std::pair{version, pyobj};
 }
 
-std::vector<VersionedItem> PythonVersionStore::batch_write_metadata(
-    std::vector<StreamId> stream_ids,
+std::vector<std::variant<VersionedItem, DataError>> PythonVersionStore::batch_write_metadata(
+    const std::vector<StreamId>& stream_ids,
     const std::vector<py::object>& user_meta,
     bool prune_previous_versions) {
-    std::vector<std::pair<VersionedItem, py::object>> results;
-    std::vector<AtomKey> keys_to_read;
-    auto stream_update_info_vector_fut = batch_get_latest_undeleted_version_and_next_version_id_async(store(),
-                                                                                                      version_map(),
-                                                                                                      stream_ids);
-    auto stream_update_info_vector = folly::collect(stream_update_info_vector_fut).via(&async::io_executor()).get();
-    std::vector<folly::Future<AtomKey>> fut_vec;
+    std::vector<arcticdb::proto::descriptors::UserDefinedMetadata> user_meta_proto_vec;
```

Comment:
`_vec` suffix not needed

Reply:
Fox maximum efficiency, you can `reserve` `user_meta.size()`, and use `emplace_back` rather than `push_back` below

Reply:
agree x2